docs/storage-config.md

```diff
@@ -0,0 +1,267 @@
+different PVCs to each libvirtd POD it creates. Thus in the short term, to make
+use of this feature would require the cloud admin to manually create libvirtd
+POD specs for each compute node, instead of using a DeamonSet resource. It is
+anticipated that either DaemonSet would be enhanced eventually, or replaced by
+a more advanced resource that can dynamically assigns PVCs to each POD.
+
+When configuring the VM spec disk
+
+```yaml
+    disks:
+      - source:
+          pvc:
+	    claimname: some-pvc-name
+	access-mode: read-only
+	copy:
+	  mode: copy-on-write
+          usage-type: fast
+```
+
+The 'usage-type' attribute refers to the 'kubevirt.io/sotrage.usage-type'
+annotations provided by the libvirtd POD. Either all libvirtd PODs would need
+to support the same annotations, or the scheduler will need to be able to only
+place VMs on the nodes where the libvirtd POD has the neccessary annotations.
+
+Each of the volume mounts provided for emphemeral storage would need to have a
+libvirt storage pool associated with them. To create the ephemeral disks, the
+virt-handler would thus use the storage pool APIs in libvirtd to create disks
+whose backing store refers to the master PVC.
+
+
+## Populating PVCs
+
+The virtual machine startup code path should not care how the PVCs associated
+with the disks are populated. It is assumed that the PVCs have been setup ahead
+of time, either by the cloud administrator, or by the tenant user. There are a
+number of possible approaches that could be used to populate these PVCs, the
+choice of which is opaque to virt-handler, where merely needs the PVC name.
+
+This doc will outline some high level concepts, but detailed technical designs
+for them will be left for separate documents to define.  The PODs described
+below could either be things that are run prior to VM launch, to statically
+populate some PVCs with persistent data, or they could be used as part of the
+dynamic disk source described above, to create storage on-demand during VM
+startup.
+
+### HTTP upload
+
+A "HTTP upload" POD could be created that references the PVC that needs to
+be populated. This POD would run a self-contained service that just exposes a
+HTTP service that accepts a PUT request at a well known URL. Any data sent to
+this URL would be immediately written into the PVC. Thus a tenant user can
+populate an arbitrary PVC with data by simply launching a POD that references
+the this 'kubevirt-pvc-upload' image, and a PVC, then use 'curl' to PUT the
+payload.
+
+### Container repository copy
+
+A "contanier repository copy" POD could be created that references the PVC that
+needs to be populated. This POD would run a self-contained service that just
+pulls down a named image from a container repository, extracts a disk image file
+(raw / qcow2 / whatever) from this image, and copies it into the PVC. When the
+copy is complete the POD would exit
+
+
+### Container repository stream
```

Comment:
And this concept does not fit into k8s. There are a few things you can do, when you want dynamic provisioning of PVCs, which fit much better into k8s. Moving this kind of generalized dynamic provision as our main entrypoint to virt-handler does not make sense in a k8s context.
The layer in virt-handler you introduce with your rest API does two things:
 * Things which virt-handler can do best directly  are moved away from virt-handler (e.g. copying an image from somewhere to the local machine), 
 * Things which are better done on the cluster side don't fit good into virt-handler flows, are now at least indirect the responsibility of virt-handler (e.g. triggering the run of a provision pod through the above rest service).

Like the kubelet, virt-handler is and should also in the future be designed, to either do host-related things directly ( everything host related: managing the VM object, mounting, copying, ... ), or wait until preconditions are not met (e.g. if a disk not provisioned, it should just try again later).